internal/db/dispatchRepo.go

```diff
@@ -0,0 +1,98 @@
+package db
+
+import (
+	"context"
+	"fmt"
+	"strings"
+	e "subscription-api/internal/entities"
+	"subscription-api/internal/services"
+)
+
+type DispatchRepo struct{}
+
+func NewDispatchRepo() *DispatchRepo {
+	return &DispatchRepo{}
+}
+
+const getDispatchByIdQ string = `
+	select cd.u_id, cd.label, cd.template_name, cd.base_currency , cd.target_currencies, cd.send_at, count(cs.user_id) subs_count
+	from subs."currency_dispatches" cd
+	left join subs."currency_subscriptions" cs
+	on cs.dispatch_id = cd.id
+	where cd.u_id = $1
+	group by cd.id;
+`
+
+func (r *DispatchRepo) GetDispatchByID(ctx context.Context, db DB, dispatchId string) (e.CurrencyDispatch, error) {
```

Comment:
Maybe we should make db depenedency of DispatchRepo?
Now this is seems  more like functional programming

Reply:
It was made intentionally to make queries to multiple repositories in single transaction

Reply:
Like here (one transaction contains 3 requests to different repositories):

```go
func (s *dispatchService) SubscribeForDispatch(ctx context.Context, email, dispatchId string) error {
	err := s.store.WithTx(ctx, func(d db.DB) error {
		_, err := s.dispatchRepo.GetDispatchByID(ctx, d, dispatchId)
		if err != nil {
			return err
		}

		if err = s.userRepo.CreateUser(ctx, d, email); err != nil && !errors.Is(err, services.UniqueViolationErr) {
			return err
		}

		return s.subRepo.CreateSubscription(ctx, d, db.SubscriptionData{Email: email, Dispatch: dispatchId})
	})
	return err
}

```

Reply:
Ah ok I see.  Transactions in application layer is very tricky question.
Ok, let's left as it is by now.